src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,174 @@
+        this.overlapRate = parseDoubleParameter(parameters, OVERLAP_RATE_FIELD, DEFAULT_OVERLAP_RATE);
+        this.tokenizer = parseStringParameter(parameters, TOKENIZER_FIELD, DEFAULT_TOKENIZER);
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param runtimeParameters a map for runtime parameters, containing the following runtime parameters:
+     * 1. max_token_count the max token limit for the tokenizer
+     */
+    @Override
+    public List<String> chunk(final String content, final Map<String, Object> runtimeParameters) {
+        // before chunking, validate and parse runtimeParameters
+        validatePositiveIntegerParameter(runtimeParameters, MAX_TOKEN_COUNT_FIELD, DEFAULT_MAX_TOKEN_COUNT);
+        int maxTokenCount = parseIntegerParameter(runtimeParameters, MAX_TOKEN_COUNT_FIELD, DEFAULT_MAX_TOKEN_COUNT);
+
+        List<AnalyzeToken> tokens = tokenize(content, tokenizer, maxTokenCount);
+        List<String> chunkResult = new ArrayList<>();
+
+        int startTokenIndex = 0;
+        int startContentPosition, endContentPosition;
+        int overlapTokenNumber = (int) Math.floor(tokenLimit * overlapRate);
+
+        while (startTokenIndex < tokens.size()) {
+            if (startTokenIndex == 0) {
+                // include all characters till the start if no previous passage
+                startContentPosition = 0;
+            } else {
+                startContentPosition = tokens.get(startTokenIndex).getStartOffset();
+            }
+            if (startTokenIndex + tokenLimit >= tokens.size()) {
+                // include all characters till the end if no next passage
+                endContentPosition = content.length();
+                chunkResult.add(content.substring(startContentPosition, endContentPosition));
+                break;
+            } else {
+                // include gap characters between two passages
+                endContentPosition = tokens.get(startTokenIndex + tokenLimit).getStartOffset();
+                chunkResult.add(content.substring(startContentPosition, endContentPosition));
+            }
+            startTokenIndex += tokenLimit - overlapTokenNumber;
+        }
+        return chunkResult;
+    }
+
+    private List<AnalyzeToken> tokenize(final String content, final String tokenizer, final int maxTokenCount) {
+        AnalyzeAction.Request analyzeRequest = new AnalyzeAction.Request();
+        analyzeRequest.text(content);
+        analyzeRequest.tokenizer(tokenizer);
+        try {
+            AnalyzeAction.Response analyzeResponse = analyze(analyzeRequest, analysisRegistry, null, maxTokenCount);
+            return analyzeResponse.getTokens();
+        } catch (IOException e) {
+            throw new IllegalStateException(
+                String.format(Locale.ROOT, "%s algorithm encounters exception in tokenization: %s", ALGORITHM_NAME, e.getMessage()),
```

Comment:
what is the action from user side when they see this exception?

Reply:
In fact, the only exception here is the token number exceeds the **max_token_count**. User can simply enlarge this value.

Reply:
I don't agree on this. This is pretty generic exception message. Please reword this with the action that user has to take otherwise we will start getting questions around this when this exception happen.

Reply:
Let's change the error message to: `Token algorithm [%s] encounters exception in tokenization, please check the analyzer configuration of your index`.

Reply:
In fact, the tokenization analyzer method will never generate checked exception. We may only encounter runtime exception. I cannot think of another other possible exception besides exceeding **max_token_count**. That's why I decide to include the origin exception message.